dex/src/state.rs

```diff
@@ -1554,7 +1565,15 @@ pub(crate) mod account_parser {
                 remaining_accounts,
             ) = array_refs![accounts, 5, 2, 2, 1; .. ;];
 
-            let market_authority = remaining_accounts.first();
```

Comment:
An alternative to this would be to create a new `InitializePermissionedMarket` instruction.